packages/wellknown-did-issuer/src/agent/WellKnownDidIssuer.ts

```diff
@@ -0,0 +1,239 @@
+    const addServices = args.issuances.map((issuance: IIssueDomainLinkageCredentialArgs) =>
+      this.addLinkedDomainsService(
+        {
+          did: issuance.did,
+          origin: issuance.origin,
+          serviceId: issuance.serviceId,
+        },
+        context
+      )
+    )
+
+    return Promise.all(addServices).then(async () =>
+      new Issuer()
+        .issueDidConfigurationResource({
+          issuances: await this.mapIssuances(args.issuances),
+          issueCallback:
+            typeof args.credentialIssuance === 'string' ? await this.getCredentialIssuance(args.credentialIssuance) : args.credentialIssuance,
+        })
+        .then(async (didConfigurationResource: IDidConfigurationResource) => {
+          if (args.save) {
+            // TODO add support for multiple origins when we upgrade Veramo version
+            await this.saveDidConfigurationResource({ origin: args.issuances[0].origin, didConfigurationResource }, context)
+          }
+          return didConfigurationResource
+        })
+        .catch((error: Error) => Promise.reject(Error(`Unable to issue DID configuration resource. Error: ${error.message}`)))
+    )
+  }
+
+  /** {@inheritDoc IWellKnownDidIssuer.saveDidConfigurationResource} */
+  public async saveDidConfigurationResource(args: ISaveDidConfigurationResourceArgs, context: RequiredContext): Promise<void> {
+    const didConfigurationEntity = {
+      origin: args.origin,
+      context: args.didConfigurationResource['@context'],
+      linkedDids: args.didConfigurationResource.linked_dids.map((credential: DomainLinkageCredential) =>
+        createCredentialEntity(this.normalizeCredential(credential))
+      ),
+    }
+
+    await (await this.dbConnection).getRepository(DidConfigurationResourceEntity).save(didConfigurationEntity, { transaction: true })
+  }
+
+  /** {@inheritDoc IWellKnownDidIssuer.getDidConfigurationResource} */
+  public async getDidConfigurationResource(args: IGetDidConfigurationResourceArgs, context: RequiredContext): Promise<IDidConfigurationResource> {
+    const result = await (await this.dbConnection).getRepository(DidConfigurationResourceEntity).findOne({
+      where: { origin: args.origin },
+      relations: this.didConfigurationResourceRelations,
+    })
+
+    if (!result) {
+      return Promise.reject(Error(`No DID configuration resource found for origin: ${args.origin}`))
+    }
+
+    return didConfigurationResourceFrom(result)
+  }
+
+  /** {@inheritDoc IWellKnownDidIssuer.issueDomainLinkageCredential} */
+  public async issueDomainLinkageCredential(args: IIssueDomainLinkageCredentialArgs, context: RequiredContext): Promise<DomainLinkageCredential> {
+    const did: string = parseDid(args.did).did
+
+    if (new URL(args.origin).origin !== args.origin) {
+      return Promise.reject(Error('Origin is not valid'))
```

Comment:
Maybe add the origin values, and change the error to something like 'didn't match'

Reply:
added values to the messages